### PR TITLE
Feature drag implant organ

### PR DESCRIPTION
**Pull request by elliedangerous**
[CLEAN][FEATURE][REWORK][INPUT]

## About the PR

TLDR - drag and drop organs for surgery, improvements to brain and skull surgery, refactor organs code, refactor surgery code, fixed up look-at descriptions based on op_stages

Main goal is to allowing dragging and dropping organs to a body to insert them during surgey (enables transplants for mediborg). Unfortunetly this PR got away from me so there more in here than i'd like in one PR. But it's all related to the surgery system and to each other. I just kept finding more things that i wanted to finish for the feature to be complete.

For all of the code/obj/item/organ files. I'm using inheritance to consolidate duplicate code. The organ parent /obj/item/organ has two new procs that are called from attack() and directly in the case of a drag event. One's a check if we can attach an organ, and the other does the attaching. Some child organs with special cases will override these functions (brain, lung, heart, kidney, eye) and I needed to impliment some basic things on Skull and Butt since they are not descendants of organ (for some reason). Finally added MouseDrop to mob/living that does a type check for organs, and calls these. I placed it low enough in the inheritance tree that in theory you could surgery other things with an OrganHolder in the future.

Paired organs (eyes) will randomly pick a slot when dragged and may require more than one attempt until it hits the right slot. In game this is kinda like just tossing the organ in the right direction then prodding it into mostly the right place. Since you're not bothering to actually pick it up, it seems fair.

I had to rework surgery op_stages for Heads, Skulls, Brains
To get drag and drop brains working I needed to fix up re-inserting a brain, and decided to fully impliment skulls as well as you shouldn't be able to put a brain back if there's no skull. Previously you needed to go down the head removal path to advance head op_stage enough to reinsert a brain, and that's silly. I added OrganHolder.Head.scalp_op_stage to track a cut across the top of the head as it's own thing. Non ideal, but could not think of a better place to put it. Head is too complext to do something like chest op_stage in my opinion. And since the var is on the head object it will go with the head for head transplants.

Odds and ends also changed:
+ Descriptions when looking at someone based on their op_stage and present/missing organs
+ Descriptions of a decapitated head when looking at it based on it's op_stages and present/missing organs
+ Fix organHolder to tell the head when organs are missing since the organs go with it when decapitated, and update body correctly when a head is reattached with organs still in it.
+ Added stapler to mediborg toolkit
+ Because i'm bypassing attack() for bio limb drag, i need to reimpliment some check in the organ attach() function

Final thought. People are likely going to say "if organs, why not pills? Why not X or Y" and ask for more until we slip n' slide into just getting rid of the 'borg's don't have hands' restriction. I have no answer to that, so i've put the input tag on this PR

## Why's this needed?

This is intended to complete the surgery code for mediborg. Before this patch borg can remove organs, but cannot transplant organs since they cannot pick them up.

## Changelog
```
(u)Uta
(*)You can now transplant organs via drag and drop, including mediborgs. L/R organs will randomly pick a side. Can only drag organs, so you still need a human doc to staple that nuke disc to your bleeding stump arm.
(*)Skulls can now be re-inserted into an empty head. A skull must be present for a brain to be inserted into a human head.
(*)Descriptions via look updated for current operation status. Same for decapitated heads.
(+)Added stapler to medical borg module.
```

### CODE REVIEW

**Comment by MarkNais** [comment]
Sweet.
Do borgs have a hemeostat, and does it "work" while other tools are in use?

**Comment by elliedangerous** [comment]
those oldest two commits should already be merged

**Comment by elliedangerous** [comment]
Yes they have them, and yes they work from any slot

**Comment by elliedangerous** [comment]
ya know, just occurred to me that it might be fun and balance to have a chance to miss the toss. I would be hilarious to see a borg missing over an over throwing a liver at someone on the op table as they slowly die of blood poisioning

File: code/modules/medical/surgery_procs.dm [hunk 1]
```diff
@@ -1174,68 +1227,6 @@
 		else
 			return 0
 
-
```
Comment (elliedangerous) [inline code/modules/medical/surgery_procs.dm:1177]: there are a couple 'heart.op_stage' red sections in the PR. I pulled them as heart had already been converted to chest.op_stage

**Comment by Tiggersaurus** [comment]
Talking as a Borg main (so obvious bias) this would be amazing, I love you. So many times I've helped remove someone's dying liver only to have to try+keep them alive while I yell for someone with hands to insert a new one.

Helping replace lost limbs would also be amazing as again when the shit hits the fan, I can get limbs back on while doctors triage others rather than some poor bastard having to wait around for goodness knows how long for a spare doctor.

As for the chance to miss, I just have an image of two Borg playing ping pong with someone's kidney as they both keep missing 🤭

File: code/modules/medical/surgery_procs.dm (hunk 1 above)
Comment (Kyle2143) [reply code/modules/medical/surgery_procs.dm:1177]: Hah, that's good. When I added the new organs, I just left the heart as-is and didn't much about with it much. I figured it would be better to put it in the check op stage, but didn't really care to go through whatever it would take to make it fit.

File: code/obj/item/organs/appendix.dm [hunk 2]
```diff
@@ -1,40 +1,12 @@
 /obj/item/organ/appendix
 	name = "appendix"
 	organ_name = "appendix"
+	organ_holder_name = "appendix"
+	organ_holder_location = "chest"
+	organ_holder_required_op_stage = 3.0
 	icon_state = "appendix"
 	failure_disease = /datum/ailment/disease/appendicitis
 
-	attack(var/mob/living/carbon/M as mob, var/mob/user as mob)
```
Comment (Kyle2143) [inline code/obj/item/organs/appendix.dm:7]: Removing all these attack procs that are basically duplicates are good. Not sure why I didn't think of doing that.

**Comment by Kyle2143** [comment]
At a glance, these look like good changes. Definitely a good deal of cleanliness things like the new proc attach_organ which removes a ton of repeat code. 

The labels list this as a rework, but I don't really see how though. Feature wise, the main additions seem to be allowing borgs to insert organs, and changing skull surgery. Unless you used the rework tag to mean that a lot of code was reworked, which is kinda what the cleanliness tag is used for.

**Comment by elliedangerous** [comment]
Perhaps i was misusing the rework tag, but i was thinking of moving the head and skull op_stages and the decent number of changes to descriptions and surgery_procs when I added it.

Enough of the attack functions had to override that I could see it going either way. I just personally will err on the side of removing dup code in most cases.

**Comment by flrsh** [comment]
Hi, could you elaborate on the `Descriptions via look updated for current operation status. Same for decapitated heads.` changelog entry? Is it saying that there are now additional examine messages for people, depending on their current surgery progress? If so, which ones, besides for decapitating heads?

**Comment by elliedangerous** [comment]
Yes, some of the surgery descriptions when you look at someone were triggering when they should not, and I added more. Looking at a body is the only way to get a hint of what op_stage you're at so i went heavy on them.

t_his is just the pronoun var for this function.

GENERAL
"[src.name] no longer has a skull in [t_his] head, [t_his] face is just empty skin mush!" existed but was moved under a check to see if you can see the person's face

SCALP_OP_STAGE - hidden behind 'can see head' checks for masks
"[src.name] has an open incision on [t_his] head!" existed, but now for low scalp_op_stage
"[src.name]'s head has been cut open!" for scalp_op_stage at the brain removal level
"[src.name]'s head has been cut open and [t_his] brain is gone!" for scalp_op_stage at the brain removal level, brain removed
"There's a gaping hole in [src.name]'s head!" for scalp_op_stage at the skull removal level, but the brain has been put back
"There's a gaping hole in [src.name]'s head and [t_his] brain is gone!" for scalp_op_stage at the skull removal level and the brain is gone
"There's a gaping hole in [src.name]'s head and [t_his] skull is gone!" for scalp_op_stage at the skull removal level and a missing skull (fires instead of brain if both are gone)

HEAD.OP_STAGE - hidden behind 'can see head' checks for masks
"[src.name] has a huge incision across their neck!" at the beginning of head removal path
"[src.name]'s head is barely attached!" just before decapitation
"[src.name] has been decapitated!" existing

CHEST.OP_STAGE - altered when i cleaned up heart.op_stage
"[src.name] has an open incision on [t_his] chest!" removed as it was for low heart.op_stage and we have another for low chest op_stage
"[src.name]'s chest is cut wide open!" For the op_stage right before a heart is removed (or after it's re-added)
"[src.name]'s chest is cut wide open and [t_his] heart has been removed!" for an open chest and missing heart


The descriptions for a decapitated head object were also altered, but are fundamentally the same as scalp_op_stages above


Finally the tri-message descriptions in surgery_procs were also altered when re-opening. If you suture up a head with no brain and no skull the messages were changed to reflect that they are already gone when you cut into it again. but those are kinda long to elaborate in detail here.
